--- /usr/lib64/ruby/gems/1.8/gems/geokit-rails-1.1.4/lib/geokit-rails/acts_as_mappable.rb.orig  2011-05-09 08:42:18.889096820 -0700
+++ /usr/lib64/ruby/gems/1.8/gems/geokit-rails-1.1.4/lib/geokit-rails/acts_as_mappable.rb   2011-05-09 08:44:04.977148386 -0700
@@ -30,7 +30,21 @@
     
     # Class method to mix into active record.
     module ClassMethods # :nodoc:
-      
+
+# Hard-copied from http://api.rubyonrails.org/classes/ActiveRecord/Base.html#M002310
+      def merge_conditions(*conditions)
+        segments = []
+
+        conditions.each do |condition|
+          unless condition.blank?
+            sql = sanitize_sql(condition)
+            segments << sql unless sql.blank?
+          end
+        end
+
+        "(#{segments.join(') AND (')})" unless segments.empty?
+      end
+
       # Class method to bring distance query support into ActiveRecord models.  By default
       # uses :miles for distance units and performs calculations based upon the Haversine
       # (sphere) formula.  These can be changed by setting Geokit::default_units and

